src/System.IO.FileSystem/src/System/IO/Win32FileStream.cs

```diff
@@ -1485,19 +1408,18 @@ private FileStreamAsyncResult BeginWriteAsync(byte[] array, int offset, int numB
                 if (_writePos == 0) _buffer = new byte[_bufferSize];
                 Buffer.BlockCopy(array, offset, _buffer, _writePos, numBytes);
                 _writePos += numBytes;
-
-                // Return a synchronous FileStreamAsyncResult
-                return FileStreamAsyncResult.CreateBufferedReadResult(numBytes, userCallback, stateObject, true);
+                
+                return Task.FromResult(numBytes);
```

Comment:
You should be able to just return a singleton Task here, e.g. Task.CompletedTask.